packages/components/nodes/memory/UpstashRedisBackedChatMemory/UpstashRedisBackedChatMemory.ts

```diff
@@ -0,0 +1,131 @@
+import { INode, INodeData, INodeParams } from '../../../src/Interface'
+import { getBaseClasses } from '../../../src/utils'
+import { ICommonObject } from '../../../src'
+import { BufferMemory, BufferMemoryInput } from 'langchain/memory'
+import { UpstashRedisChatMessageHistory, UpstashRedisChatMessageHistoryInput } from "langchain/stores/message/upstash_redis"
+import { Redis, RedisConfigNodejs } from '@upstash/redis'
+
+class UpstashRedisBackedChatMemory_Memory implements INode {
+    label: string
+    name: string
+    version: number
+    description: string
+    type: string
+    icon: string
+    category: string
+    baseClasses: string[]
+    inputs: INodeParams[]
+
+    constructor() {
+        this.label = 'Upstash Redis-Backed Chat Memory'
+        this.name = 'UpstashRedisBackedChatMemory'
+        this.version = 1.0
+        this.type = 'UpstashRedisBackedChatMemory'
+        this.icon = 'upstash.svg'
+        this.category = 'Memory'
+        this.description = 'Summarizes the conversation and stores the memory in upstash Redis server'
+        this.baseClasses = [this.type, ...getBaseClasses(BufferMemory)]
+        this.inputs = [
+            {
+                label: 'Base URL',
+                name: 'baseURL',
+                type: 'string',
+                default: 'redis://localhost:6379'
+            },
+            {
+                label: 'Token',
```

Comment:
Can we put the token to credential file?
Here is an example PR - https://github.com/FlowiseAI/Flowise/pull/882/files

Reply:
Acknowledged